tests/test_RBAC.py

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+
+import pytest
+from ocean_lib.common.agreements.service_types import ServiceTypes
+from ocean_lib.models.data_token import DataToken
+
+from ocean_provider.constants import BaseURLs
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.accounts import generate_auth_token
+from ocean_provider.validation.algo import build_stage_output_dict
+from ocean_provider.validation.requests import RBACValidator
+from tests.helpers.compute_helpers import build_and_send_ddo_with_compute_service
+from tests.test_helpers import (
+    get_dataset_ddo_with_access_service,
+    mint_tokens_and_wait,
+    send_order,
+)
+
+
+def test_invalid_request_name():
+    req = {"document": "My Doc"}
+    with pytest.raises(RequestNotFound) as err:
+        RBACValidator(request_name="MyRequest", request=req)
+    assert err.value.args[0] == "Request name is not valid!"
+
+
+encrypt_endpoint = BaseURLs.ASSETS_URL + "/encrypt"
+
+
+def test_encrypt_request_payload():
+    document = {
+        "url": "http://localhost:8030" + encrypt_endpoint,
+        "index": 0,
+        "checksum": "foo_checksum",
+        "contentLength": "4535431",
+        "contentType": "text/csv",
+        "encoding": "UTF-8",
+        "compression": "zip",
+    }
+    req = {"document": json.dumps(document)}
+    validator = RBACValidator(request_name="EncryptRequest", request=req)
+    payload = validator.build_payload()
+    assert validator.request == req
+    assert payload["eventType"] == "encryptUrl"
+    assert payload["component"] == "provider"
+    assert payload["credentials"] == {
+        "type": "address",
+        "address": validator.provider_address,
+    }
+
+
+def test_initialize_request_payload(client, publisher_wallet, consumer_wallet):
+    ddo = get_dataset_ddo_with_access_service(client, publisher_wallet)
+    dt_contract = DataToken(ddo.data_token_address)
+    sa = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    mint_tokens_and_wait(dt_contract, consumer_wallet, publisher_wallet)
+
+    document = {
+        "documentId": ddo.did,
+        "serviceId": sa.index,
+        "serviceType": sa.type,
+        "dataToken": ddo.data_token_address,
+        "consumerAddress": consumer_wallet.address,
+    }
+
+    req = {"document": json.dumps(document)}
+    validator = RBACValidator(request_name="InitializeRequest", request=req)
+    payload = validator.build_payload()
+    assert validator.request == req
+    assert payload["eventType"] == "initialize"
+    assert payload["component"] == "provider"
+    assert payload["credentials"] == {
+        "type": "address",
+        "address": validator.provider_address,
+    }
+    assert payload["dids"][0]["did"] == validator.get_dids(sa.index)[0]["did"]
```

Comment:
You are again checking that a = a. Use the did that you just defined a few lines above this.

Reply:
Please extend this comment to the usage of serviceId etc. And change accordingly for all the tests file.